hash_to_array.rb

```diff
@@ -0,0 +1,26 @@
+def hash_to_array(hash)
+    puts "Keys are "
+    puts hash.keys
+    puts "Values are "
+    puts hash.values
+end
+
+def read_hash(index,hash)
+    puts("Enter #{index}th key")
+    hash_key = gets.chomp
+    puts("Enter #{index}th value")
```

Comment:
Since you're using ordinals, you need to be careful with the index; once `hash` is populated, `hash.keys.first` is actually equal to `hash.keys[0]` instead of `hash.keys[1]` in general, so I would start with this:

```suggestion
    puts("Enter #{index + 1}th key")
    hash_key = gets.chomp
    puts("Enter #{index + 1}th value")
```

The other problem is that English [ordinal numerals](https://en.wikipedia.org/wiki/Ordinal_numeral) have special suffixes for certain numbers; for example, "first, second, and third" are written with digits as "1st, 2nd, and 3rd" instead of "1th, 2th, and 3th". While you could write special logic to handle these cases, it's simpler to just avoid ordinals like so:

```suggestion
    puts("Enter key ##{index + 1}")
    hash_key = gets.chomp
    puts("Enter value ##{index + 1}")
```